bin/fuel-core/src/cli/snapshot.rs

```diff
@@ -69,19 +73,25 @@ pub async fn exec(command: Command) -> anyhow::Result<()> {
     let db = Database::new(std::sync::Arc::new(data_source));
 
     match command.subcommand {
-        SubCommands::Everything { chain_config } => {
-            let config: ChainConfig = chain_config.parse()?;
-            let state_conf = StateConfig::generate_state_config(db)?;
-
-            let chain_conf = ChainConfig {
-                initial_state: Some(state_conf),
-                ..config
+        SubCommands::Everything {
+            chain_config,
+            output_dir,
+        } => {
+            let chain_params = match chain_config.as_str() {
+                LOCAL_TESTNET => ChainConfig::local_testnet(),
+                _ => ChainConfig::load_from_directory(&chain_config)?,
             };
+            let chain_state = StateConfig::generate_state_config(db)?;
 
-            let stdout = std::io::stdout().lock();
+            std::fs::create_dir_all(output_dir)?;
+            let parameters_writer = File::create("chain_parameters.json")?;
```

Comment:
These strings should be constants for easier identification and later refactoring.

```suggestion
            let parameters_writer = File::create("CHAIN_PARAMS_FILENAME")?;
```